### PR TITLE
Add a content entity form which allows to set revisions

### LINKED ISSUE #24: Provide a entity form which has some of the revision logic built in
`\Drupal\content_entity_base\Entity\Form\EntityBaseForm` makes it possible to have a node like edit form, which has the revision information checkboxes. Let's provide something like that in entity module as well

### PR DESCRIPTION
**Pull request by dawehner**
Fixes #24

### CODE REVIEW

File: src/Form/ContentEntityFormWithRevisions.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    if ($bundle_entity instanceof RevisionableEntityBundleInterface) {
+      // Always use the default revision setting.
+      $this->entity->setNewRevision($bundle_entity && $bundle_entity->shouldCreateNewRevision());
+    }
+  }
+
+  protected function getBundleEntity() {
+    $bundle_entity = $this->entity->{$this->entity->getEntityType()->getKey('bundle')}->referencedEntities()[0];
```
Comment (bojanz) [inline src/Form/ContentEntityFormWithRevisions.php]: This line really needs to be broken down into several :) the method is also missing docs

File: src/Form/ContentEntityFormWithRevisions.php [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    return $bundle_entity;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state) {
+
```
Comment (bojanz) [inline src/Form/ContentEntityFormWithRevisions.php]: The empty lines between these three variables are not needed.

File: src/Form/ContentEntityFormWithRevisions.php [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      if ($this->entity->getEntityType()->hasLinkTemplate('collection')) {
+        $form_state->setRedirectUrl($this->entity->toUrl('collection'));
+      }
+      else {
+        $form_state->setRedirectUrl($this->entity->toUrl('canonical'));
+      }
+    }
+    else {
+      // In the unlikely case something went wrong on save, the entity will be
```
Comment (bojanz) [inline src/Form/ContentEntityFormWithRevisions.php]: Core never cares if something went wrong on save. And this message doesn't help much. Maybe better to let it fail?

**Comment by bojanz** [comment]
Great work!

**Comment by bojanz** [comment]
Wondering about the naming of this form, and a revisionable base entity class.
ContentEntityFormWithRevisions -> RevisionableContentEntityForm / RevisionableEntityForm?

File: src/Form/ContentEntityFormWithRevisions.php (hunk 2 above)
Comment (dawehner) [inline src/Form/ContentEntityFormWithRevisions.php]: Done

File: src/Form/ContentEntityFormWithRevisions.php (hunk 3 above)
Comment (dawehner) [inline src/Form/ContentEntityFormWithRevisions.php]: Beside it does, see `\Drupal\node\NodeForm::save` and `\Drupal\block_content\BlockContentForm::save`

**Pull request merged**